### DISCUSSION TITLE
DataGrid on a TabView not appearing in macOS but does appear in iOS. What did I do wrong?

### DISCUSSION THREAD

**Discussion by roberuc** [unoplatform/uno#8105] [Q&A]
Smallest description of problem: I am using "Visual Studio 2019 for Mac".
(I did install "Visual Studio 2022 Preview for Mac" but later deleted it. So could be some residue problem.)
I created a template solution using the CLI (in the "Documents" folder):  dotnet new unoapp -o AU_new
Deleted the "global.json" file.
Removed a lot of Projects because I am just interested in the Shared, Mac, and iOS. (Kept just those.)

Did a Nuget restore on the solution. Set the StartUp Project to MacOS.
Had to add a couple of Nuget packages to both the macOS and iOS projects:

       Microsoft.UI.Xaml
       Uno.Microsoft.Toolkit.Uwp.UI.Controls.DataGrid

edited the MainPage.xml to this: 

 ```
<Page
    x:Class="AU_new.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:AU_new"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"mc:Ignorable="d"
    xmlns:uc="using:Microsoft.UI.Xaml.Controls"
    xmlns:ud="using:Microsoft.Toolkit.Uwp.UI.Controls"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <StackPanel>
        <uc:TabView>
            <uc:TabViewItem Header="A">
                <StackPanel Background="Yellow">
                <TextBlock>TabView A</TextBlock>
                </StackPanel>
            </uc:TabViewItem>
            <uc:TabViewItem Header="B">
                <StackPanel Background="Green">
                <TextBlock>TabView Ba</TextBlock>
                <ud:DataGrid Name="gridHere" SelectionMode="Extended" AutoGenerateColumns="True" Height="200" Width="600" Foreground="Black" Background="White">
                    <ud:DataGrid.Columns>
                        <ud:DataGridTextColumn Width="150" Header="col A" Foreground="Black" />
                        <ud:DataGridTextColumn Width="150" Header="col B"  />
                        <ud:DataGridTextColumn Width="150" Header="col C"  />
                        <ud:DataGridTextColumn Width="150" Header="col D"  />
                    </ud:DataGrid.Columns>
                </ud:DataGrid>
                <TextBlock>TabView Bz</TextBlock>
                </StackPanel>
            </uc:TabViewItem>
        </uc:TabView>
    </StackPanel>
</Page>
```

Compiles for macOS and iOS. Gives some warnings but it runs.
It works on iOS. In 2nd tab, the DataGrid appears with 4 columns. (Yeah!)
But on macOS, the Tab items appear but the DataGrid ends up as a big empty block of green background color.
(Side note: I can take this to a PC running Visual Studio 2019 and put this code into an Uno App, and it runs there for UWP.
I do not bring the entire solution there. Seems there are some subtle differences between the various Op Sys Projects.
But the shared common folder can be copied.  That is so freaking cool!)
Really curious what I am doing wrong on macOS?

**Comment by jeromelaban** [comment]
Thanks for the report. It's likely that you're hitting this issue: https://github.com/unoplatform/uno/issues/6375

**Reply by roberuc** [reply]
Thanks. #6375 says it's a problem with Binding to data. My sample is doesn't even have Binding... unless there is some empty Binding by default. But once I get empty columns appearing, that is where I would be going next. And my problem happens even if the TabView elements are removed. I tried playing with earlier versions of various NuGet packages to see if it worked in a previous incarnation... but no luck yet on macOS.

**Reply by jeromelaban** [reply]
Thanks. I've opened https://github.com/unoplatform/uno/issues/8110 for tracking.

**Reply by roberuc** [reply]
Yuck.
Found an ugly work around but is not as elegant or pretty as a DataGrid. And have not figured out yet how to implement a column sort. (Besides being new at Mac, I am also new at UWP.)
I'm guessing it may be a while until this gets fixed as #6375 has been around since 7/1/2021.

"MyFakeData" is the name of my public class holding the data.
And "listMFD_C" is a: private List<MyFakeData> listMFD_C; // Made in MainPage.xaml.cs

    <ListView Name="gridCompleted" HorizontalAlignment="Left" VerticalAlignment="Center"  >
        <Grid>
            <Grid.ColumnDefinitions>
                <ColumnDefinition Width="150" />
                <ColumnDefinition Width="150" />
                <ColumnDefinition Width="150" />
                <ColumnDefinition Width="200" />
                <ColumnDefinition Width="150" />
            </Grid.ColumnDefinitions>
            <TextBlock Grid.Column="0" Text="{x:Bind local:App.Strings('Lbl_Grid_Sender')}" />
            <TextBlock Grid.Column="1" Text="{x:Bind local:App.Strings('Lbl_Grid_Practitioner')}" />
            <TextBlock Grid.Column="2" Text="{x:Bind local:App.Strings('Lbl_Grid_Software')}" />
            <TextBlock Grid.Column="3" Text="{x:Bind local:App.Strings('Lbl_Grid_DateTime')}" />
            <TextBlock Grid.Column="4" Text="{x:Bind local:App.Strings('Lbl_Grid_Status')}" />
        </Grid>
        <ScrollViewer>
            <GridView ItemsSource="{x:Bind listMFD_C}" SelectionMode="Single">
                <GridView.ItemTemplate>
                    <DataTemplate x:DataType="local:MyFakeData">
                        <Grid >
                            <Grid.ColumnDefinitions>
                                <ColumnDefinition Width="150" />
                                <ColumnDefinition Width="150" />
                                <ColumnDefinition Width="150" />
                                <ColumnDefinition Width="200" />
                                <ColumnDefinition Width="150" />
                            </Grid.ColumnDefinitions>
                            <TextBlock Grid.Column="0" Text="{x:Bind SenderName}" Tag="SenderName"/>
                            <TextBlock Grid.Column="1" Text="{x:Bind PractitionerName}" Tag="PractitionerName"/>
                            <TextBlock Grid.Column="2" Text="{x:Bind AppName}" Tag="AppName"/>
                            <TextBlock Grid.Column="3" Text="{x:Bind CreateDate}" Tag="CreateDate"/>
                            <TextBlock Grid.Column="4" Text="{x:Bind Status}" Tag="Status"/>
                        </Grid>
                    </DataTemplate>
                </GridView.ItemTemplate>
            </GridView>
        </ScrollViewer>
    </ListView>

**Reply by jeromelaban** [reply]
Uno is primarily working on issues based on customers requests, see [our business model](https://platform.uno/blog/sustaining-the-open-source-uno-platform/).

If you want to run on macOS, you can also run the Skia/GTK head included in your application created from a template.

**Reply by roberuc** [reply]
Whoa. I did not know that. 
I thought that was just Linux.
So are you saying the files generated in the "Skia.Gtk" will run on Mac native?
Or is it that stuff will work there after I download and install some "gtk3-runtime" app for the Mac? 
(Sort of like I just did for Windows to get that to work.)

**Reply by jeromelaban** [reply]
Yes, Skia.Gtk runs anywhere Gtk runs, so Linux, mac and Windows (down to Windows 7). You'll need to [setup gtk3 for macOS](https://platform.uno/docs/articles/get-started-vsmac.html#prerequisites).
